api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -271,18 +271,18 @@ message ExternalProcessor {
   // The default value is 5000 milliseconds (5 seconds) if not specified.
   google.protobuf.Duration deferred_close_timeout = 19;
 
-  // [#not-implemented-hide:]
   // Send body to the side stream server once it arrives without waiting for the header response from that server.
   // It only works for STREAMED body processing mode. For any other body processing modes, it is ignored.
   //
   // The server has two options upon receiving a header request:
   // 1. Instant Response: Send the header response as soon as the header request is received.
   // 2. Delayed Response: Wait for the body before sending any response.
-  //   If the server chooses the second option, it has two further choices:
-  //   2.1 Separate Responses: Send the header response first, followed by separate body responses.
-  //   2.2 Combined Response: Include both the header response and the first chunk of the body response
-  //       in a single body response message, followed by the remaining body responses.
   // In all scenarios, the header-body ordering must always be maintained.
+  //
+  // With this flag enabled, the server should not set
+  // :ref:`mode_override <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.mode_override>`
+  // in the header response as the ``mode_override`` may be ignored by Envoy if it has already
```

Comment:
Should the `may` be replaced with `will`?

RE `if sent the body to the server`, does it imply that Envoy has wiggle room to decide whether to ignore or not, but it is dependent on whether it already sent the body or not?

If possible, I think it would be good to have a more deterministic description of the protocol here.

Will the following rephrase for this added text work:
```
If enabled Envoy will ignore the
:ref:`mode_override <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.mode_override>`
value that the server sends in the header response.
This is because Envoy may have already sent the body to the server, prior to processing the header response.
```

Reply:
Yeah, it's a good idea to make it deterministic. i.e, ignore mode_override in ProcessingResponse if filter has body mode as STREAMED, and send_body_without_waiting_for_header_response true.

Reply:
@tyxia @yanavlasov please let me know whether this sounds good to you.